influxql/ast.go

```diff
@@ -4300,7 +4300,13 @@ func Rewrite(r Rewriter, node Node) Node {
 		n.Fields = Rewrite(r, n.Fields).(Fields)
 		n.Dimensions = Rewrite(r, n.Dimensions).(Dimensions)
 		n.Sources = Rewrite(r, n.Sources).(Sources)
-		n.Condition = Rewrite(r, n.Condition).(Expr)
+
+		// n.Condition may be nil. In that case, Rewrite will return
+		// nil. This causes a panic in the type assertion to Expr,
+		// because nil does not satisfy the Expr interface:
+		if n.Condition != nil {
+			n.Condition = Rewrite(r, n.Condition).(Expr)
```

Comment:
This looks like it's trying to assert the return value is an expression. It seems to me this should instead by something like this:

```go
if cond := Rewrite(r, n.Condition); cond != nil {
    n.Condition = cond.(Expr)
} else {
    n.Condition = nil
}
```

This way, it's possible for the rewrite to return a nil condition. It appears that this appears to cause an error when nil is passed in because it doesn't touch the passed in nil and then tries to assert the nil is an `Expr`.